pkg/cloud/services/ec2/launchtemplate.go

```diff
@@ -962,27 +966,54 @@ func (s *Service) getFilteredSecurityGroupIDs(securityGroup infrav1.AWSResourceR
 	return ids, nil
 }
 
-func getLaunchTemplateInstanceMarketOptionsRequest(spotMarketOptions *infrav1.SpotMarketOptions) *ec2.LaunchTemplateInstanceMarketOptionsRequest {
-	if spotMarketOptions == nil {
-		// Instance is not a Spot instance
-		return nil
+func getLaunchTemplateInstanceMarketOptionsRequest(i *expinfrav1.AWSLaunchTemplate) (*ec2.LaunchTemplateInstanceMarketOptionsRequest, error) {
+	if i.MarketType != "" && i.MarketType != infrav1.MarketTypeCapacityBlock && i.SpotMarketOptions != nil {
```

Comment:
This function does the same thing as `getInstanceMarketOptionsRequest`. Sad that the distinct types don't allow it to easily be one function.

From comparing both functions, I think this is a bug:

```suggestion
func getLaunchTemplateInstanceMarketOptionsRequest(i *expinfrav1.AWSLaunchTemplate) (*ec2.LaunchTemplateInstanceMarketOptionsRequest, error) {
	if i.MarketType != "" && i.MarketType == infrav1.MarketTypeCapacityBlock && i.SpotMarketOptions != nil {
```

Reply:
ahh, thanks. fixed